src/java/com/zimbra/qa/unittest/TestSSDBEphemeralStore.java

```diff
@@ -35,298 +39,279 @@
-                    try (Jedis jedis = pool.getResource()) {
-                        jedis.flushDB();
-                    }
+        if (SSDBStoreConfigured) {
+            for (Pair<EphemeralInput, EphemeralLocation> pair: toDelete) {
+                EphemeralInput input = pair.getFirst();
+                EphemeralLocation location = pair.getSecond();
+                try {
+                    store.delete(input.getEphemeralKey(), (String) input.getValue(), location);
+                } catch (ServiceException e) {
+                    // ignore failed deletions
                 }
-            } catch (ClassCastException e) {
-                //ignore
             }
+            toDelete.clear();
+            SSDBEphemeralStore.getFactory().shutdown();
         }
     }
 
+    private void addDeletionEntry(EphemeralInput input, EphemeralLocation location) {
+        toDelete.add(new Pair<EphemeralInput, EphemeralLocation>(input, location));
+    }
+
+    private void doSet(EphemeralInput input, EphemeralLocation location) throws ServiceException {
+        store.set(input, location);
+        addDeletionEntry(input, location);
+    }
+
+    private void doUpdate(EphemeralInput input, EphemeralLocation location) throws ServiceException {
+        store.update(input, location);
+        addDeletionEntry(input, location);
+    }
+
     @Test
     public void testSetGetLastLogin() throws Exception {
-        if(SSDBStoreConfigured) { //switch to assumptions when we upgrade test suite to Junit 4
-            String firstLogin = "20160912212057.178Z";
-            EphemeralStore store = SSDBEphemeralStore.getFactory().getStore();
-            assertTrue(store instanceof SSDBEphemeralStore);
-            EphemeralLocation accountIDLocation = new EphemeralLocation() {
-                @Override
-                public String[] getLocation() { return new String[] { "account", ACCOUNT_ID }; }
-            };
-            EphemeralKey eKey = new EphemeralKey(Provisioning.A_zimbraLastLogonTimestamp);
-            EphemeralInput attr = new EphemeralInput(eKey, firstLogin);
-            store.set(attr, accountIDLocation);
-
-            EphemeralResult retAttr = store.get(eKey, accountIDLocation);
-            assertNotNull(retAttr);
-            assertEquals("Found incorrect last logon timestamp value",firstLogin, retAttr.getValue());
-        }
+        Assume.assumeTrue(SSDBStoreConfigured);
+        String firstLogin = "20160912212057.178Z";
+        EphemeralLocation accountIDLocation = new EphemeralLocation() {
+            @Override
+            public String[] getLocation() { return new String[] { "account", ACCOUNT_ID }; }
+        };
+        EphemeralKey eKey = new EphemeralKey(Provisioning.A_zimbraLastLogonTimestamp);
+        EphemeralInput attr = new EphemeralInput(eKey, firstLogin);
+        doSet(attr, accountIDLocation);
+        EphemeralResult retAttr = store.get(eKey, accountIDLocation);
+        assertNotNull(retAttr);
```

Comment:
Not new code but please use the assert variants which have strings to explain what is wrong so that time isn't wasted diagnosing why tests are failing.